src/main.zig

```diff
@@ -0,0 +1,49 @@
+const std = @import("std");
+const builtin = @import("builtin");
+const filestream = std.fs.openFileAbsolute;
+const stdoutFile = std.io.getStdOut();
+const stdout = std.io.getStdOut().writer();
+
+pub fn main() !void {
+    // .riscv32 - don't work:
+    // https://github.com/ziglang/zig/blob/c955379504d4866f9c474c50317b2a0da18ee631/lib/std/os/linux.zig#L35-L44
+    const arch = switch (builtin.cpu.arch) {
+        .x86_64 => "x86_64",
+        .aarch64 => "ARM64",
+        .riscv64 => "RISC-V",
+        .mipsel, .mips => "MIPS",
+        else => @compileError("Unsupported CPU Architecture"),
+    };
+    try stdout.print("\nLet's have a look at your shiny {s} system! :)\n\n", .{arch});
+
+    // Read HW linux info
+
+    if (builtin.os.tag != .linux) {
+        @compileError("Unsupported another OS");
+    }
+
+    const cpuinfo = try filestream("/proc/cpuinfo", .{});
+    defer cpuinfo.close();
+    try stdout.print("cpuinfo:\n", .{});
+    try stdoutFile.writeFileAll(cpuinfo, .{});
+
+    var buf: [84]u8 = undefined;
```

Comment:
The Rust implementation takes the first three lines - why this random cut?

Reply:
Yes. I had to do it manually.
Because due to the transition of zig's std api some more viable alternatives are no longer working.
I plan to fix it later, when possible.
https://github.com/ziglang/zig/wiki/Zig-Newcomer-Programming-FAQs#how-do-i-read-input-one-line-at-a-time

Reply:
Can you read the whole thing, split by `\n`, and output the first three of those?

Reply:
changed

Reply:
@orangecms, I think it brought a slightly more readable improvement regarding reading meminfo.